mne_bids/utils.py

```diff
@@ -26,9 +27,145 @@
 from mne.io.constants import FIFF
 from mne.time_frequency import psd_array_welch
 
+from mne_bids.config import BIDS_ENTITIES
 from mne_bids.tsv_handler import _to_tsv, _tsv_to_str
 
 
+class BIDSPath(object):
+    """Create a partial/full BIDS filename from its component parts.
+
+    BIDS filename prefixes have one or more pieces of metadata in them. They
+    must follow a particular order, which is followed by this function. This
+    will generate the *prefix* for a BIDS filename that can be used with many
+    subsequent files, or you may also give a suffix that will then complete
+    the file name.
+
+    Note that not all parameters are applicable to each kind of data. For
+    example, electrode location TSV files do not need a "task" field.
+
+    Parameters
+    ----------
+    subject : str | None
+        The subject ID. Corresponds to "sub".
+    session : str | None
+        The session for a item. Corresponds to "ses".
+    task : str | None
+        The task for a item. Corresponds to "task".
+    acquisition: str | None
+        The acquisition parameters for the item. Corresponds to "acq".
+    run : int | None
+        The run number for this item. Corresponds to "run".
+    processing : str | None
+        The processing label for this item. Corresponds to "proc".
+    recording : str | None
+        The recording name for this item. Corresponds to "recording".
+    space : str | None
+        The coordinate space for an anatomical file. Corresponds to "space".
+    prefix : str | None
+        The prefix for the filename to be created. E.g., a path to the folder
+        in which you wish to create a file with this name.
+    suffix : str | None
+        The suffix for the filename to be created. E.g., 'audio.wav'.
+
+    Examples
+    --------
+    >>> bids_basename = make_bids_basename(subject='test', session='two',
+                                           task='mytask', suffix='data.csv')
+    >>> print(bids_basename)
+    sub-test_ses-two_task-mytask_data.csv
+    >>> print(bids_basename.__repr__())
+    BIDSPath (sub-test_ses-two_task-mytask_data.csv)
```

Comment:
I'm wondering if the repr should actually list the different entities, something like:

```py
<BIDSPath |
prefix=None,
task='blah',
subject='blah'
>
```
Thoughts?

Reply:
I like that. Perhaps in addition:

```Python
<BIDSPath | (sub-test_ses-two_task-mytask_data.csv)
prefix=None,
task='blah',
subject='blah'
...
>
```

?